src/tanks/Tank.js

```diff
@@ -60,34 +63,73 @@ class Tank extends GameObject {
 	nitroRepairTimerLeft: number;
 
 	// parts: 
-	chassis: TankPart;
-	treads: TankPart;
+	chassis: Chassis;
 	mainGun: Gun;
-	scanner: ?Scanner;
-	jammer: ?Jammer;
-	parts: Array<?TankPart>;
+	secondaryGun: Gun;
+	scanner: Scanner;
```

Comment:
make nullable,
make it so that we only have one scanner and that scanner knows about whether it is anti-jammer, and whether it is an item scanner

Reply:
I made it so all components are necessary, however I've added error handling for what is now 'empty' components. Speak to me about this later.